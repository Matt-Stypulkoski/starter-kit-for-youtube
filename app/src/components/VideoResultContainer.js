import { React } from 'react';
import VideoResult from './VideoResult.js';

function VideoResultContainer(props) {
    return (
        <div className="results-container" id="results-container">
            <ul className="results-list">
                {props.videoList.map(function (vidElement) {
                    return <VideoResult title={vidElement[0]} views={vidElement[1]} thumbnailURL={vidElement[2]} altText={vidElement[3]} />
                })}
            </ul>
        </div>
    )
}

export default VideoResultContainer;