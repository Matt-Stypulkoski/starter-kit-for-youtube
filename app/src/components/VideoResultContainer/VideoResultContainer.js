import { React } from 'react';
import VideoResult from '../VideoResult/VideoResult.js';

function VideoResultContainer(props) {
    return (
        <div className="results-container" id="results-container">
            <ul className="results-list">
                {props.videoList.map(function (vidElement) {
                    // vidElement -> [title, views, thumbnailURL, altText, datetimePublished, videoId, channelTitle, channelId]
                    return <VideoResult key={vidElement[5]} title={vidElement[0]} views={vidElement[1]} thumbnailURL={vidElement[2]}
                        altText={vidElement[3]} datetimePublished={vidElement[4]} videoId={vidElement[5]} channelTitle={vidElement[6]} channelId={vidElement[7]} />
                })}
            </ul>
        </div>
    )
}

export default VideoResultContainer;