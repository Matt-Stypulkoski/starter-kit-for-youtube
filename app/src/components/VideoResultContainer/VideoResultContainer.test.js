import React from 'react';
import {cleanup, render} from '@testing-library/react';
import renderer from 'react-test-renderer';
import VideoResultContainer from './VideoResultContainer';

afterEach(cleanup)

it("renders", () => {
    const videoResultContainer = render(<VideoResultContainer videoList={[]}/>) 
});

it("renders when given props", () => {
    let videoProps = [
        "Test title",
        0,
        "../../test/videoResultTest.png",
        "Test Img",
        "2000-01-01T00:00:00Z",
        0,
        "Test channel",
        "0"
    ]
    const tree = renderer.create(<VideoResultContainer videoList={[videoProps]} />);
    expect(tree).toMatchSnapshot();
});