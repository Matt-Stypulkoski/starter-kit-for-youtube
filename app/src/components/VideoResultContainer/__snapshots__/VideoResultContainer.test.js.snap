// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when given props 1`] = `
<div
  className="results-container"
  id="results-container"
>
  <ul
    className="results-list"
  >
    <li
      className="video-list-element"
    >
      <div
        className="video-result-container"
      >
        <div
          className="video-link-container"
        >
          <a
            className="video-link"
            id={0}
            onClick={[Function]}
            target="_blank"
          >
            <img
              alt="Test Img"
              className="video-thumbnail"
              src="../../test/videoResultTest.png"
            />
          </a>
        </div>
        <div
          className="video-data-container"
        >
          <h3
            className="video-title"
          >
            Test title
          </h3>
          <div
            className="video-channel-views"
          >
            <h4>
              <a
                href="https://www.youtube.com/channel/0"
                target="_blank"
              >
                Test channel
              </a>
              <span>
                 - 
              </span>
              0
               views
            </h4>
          </div>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
